### PR TITLE
Add support for scientific notation to WKT format

### LINKED ISSUE #3686: ol.format.WKT doesn't recognize scientific number notation
I'm having a problem where default C# / SQL Server implementation of geography to WKT methods return small numbers in scientific number format with character E in them. I get the following error
Uncaught Error: Unexpected `E` at position 123 in `MULTIPOLYGON(((-5.7156469384940491 50.082709053271955, -5.7059626929961649 ... **9.1993293867552632E-05** 53.535345770529375, ...))).
The most popular 3rd party library for spatial data in C# creates WKT in same format.

### PR DESCRIPTION
**Pull request by marcjansen**
This PR suggests that we add support for WKTs where the coordinates are in scientific notation.

Please review and share your thoughts.

Fixes #3686

### CODE REVIEW

File: src/ol/format/wktformat.js [hunk 1]
```diff
@@ -491,12 +491,23 @@ ol.format.WKT.Lexer.prototype.nextToken = function() {
 ol.format.WKT.Lexer.prototype.readNumber_ = function() {
   var c, index = this.index_;
   var decimal = false;
+  var scientificNotation = false;
   do {
     if (c == '.') {
       decimal = true;
+    } else if (c !== undefined && c.toLowerCase() == 'e') {
```
Comment (elemoine) [inline src/ol/format/wktformat.js]: Any reason for not using `goog.isDef`, which we use anywhere else?

**Comment by elemoine** [comment]
I just added a comment regarding the use of `goog.isDef`. Otherwise looks good to me. Thanks for the thorough tests.

File: src/ol/format/wktformat.js (hunk 1 above)
Comment (gberaudo) [inline src/ol/format/wktformat.js]: What about simply testing `c == 'e' || c == 'E'`?

Comment (fredj) [inline src/ol/format/wktformat.js]: See https://github.com/openlayers/ol3/blob/master/src/ol/format/wktformat.js#L513 (`toUpperCase`): maybe the character is always `E` ? (haven't tried)

**Comment by marcjansen** [comment]
Thanks for the review, Éric. Now using `goog.isDef`. I'll merge once travis is happy.

**Comment by elemoine** [comment]
I love when people use `É` rather than `E` in my first name. Thanks for that :-)

File: src/ol/format/wktformat.js (hunk 1 above)
Comment (fredj) [inline src/ol/format/wktformat.js]: > See https://github.com/openlayers/ol3/blob/master/src/ol/format/wktformat.js#L513 >  (toUpperCase): maybe the character is always E ? (haven't tried)

It's not the case

**Comment by marcjansen** [comment]
I don't always think of it, but sometimes ...

**Pull request merged**